packages/test/test-end-to-end-tests/src/test/testApi.ts

```diff
@@ -0,0 +1,240 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { exec, execSync } from "child_process";
+import * as path from "path";
+import { existsSync, mkdirSync, unlinkSync } from "fs";
+
+// Loader API
+import { Loader } from "@fluidframework/container-loader";
+
+// ContainerRuntime API
+import { ContainerRuntime } from "@fluidframework/container-runtime";
+
+// Data Runtime API
+import { SharedCell } from "@fluidframework/cell";
+import { SharedCounter } from "@fluidframework/counter";
+import { Ink } from "@fluidframework/ink";
+import { SharedDirectory, SharedMap } from "@fluidframework/map";
+import { SharedMatrix } from "@fluidframework/matrix";
+import { ConsensusQueue } from "@fluidframework/ordered-collection";
+import { ConsensusRegisterCollection } from "@fluidframework/register-collection";
+import { SharedString, SparseMatrix } from "@fluidframework/sequence";
+import { TestFluidObjectFactory } from "@fluidframework/test-utils";
+
+// ContainerRuntime and Data Runtime API
+import { ContainerRuntimeFactoryWithDefaultDataStore, DataObject, DataObjectFactory } from "@fluidframework/aqueduct";
+
+import { lock } from "proper-lockfile";
+import * as semver from "semver";
+import { pkgVersion } from "./packageVersion";
+
+// List of package that needs to be install for past versions
+const packageList = [
+    "@fluidframework/aqueduct",
+    "@fluidframework/test-utils",
+    "@fluidframework/container-loader",
+    "@fluidframework/container-runtime",
+    "@fluidframework/cell",
+    "@fluidframework/counter",
+    "@fluidframework/ink",
+    "@fluidframework/map",
+    "@fluidframework/matrix",
+    "@fluidframework/ordered-collection",
+    "@fluidframework/register-collection",
+    "@fluidframework/sequence",
+];
+
+// Current versions of the APIs
+const LoaderApi = {
+    version: pkgVersion,
+    Loader,
+};
+
+const ContainerRuntimeApi = {
+    version: pkgVersion,
+    ContainerRuntime,
+    ContainerRuntimeFactoryWithDefaultDataStore,
+};
+
+const DataRuntimeApi = {
+    version: pkgVersion,
+    DataObject,
+    DataObjectFactory,
+    TestFluidObjectFactory,
+    dds: {
+        SharedCell,
+        SharedCounter,
+        Ink,
+        SharedDirectory,
+        SharedMap,
+        SharedMatrix,
+        ConsensusQueue,
+        ConsensusRegisterCollection,
+        SharedString,
+        SparseMatrix,
+    },
+};
+
+export type DataRuntimeApiType = typeof DataRuntimeApi;
+const resolutionCache = new Map<string, string>();
+
+function resolveVersion(requested: string) {
```

Comment:
can we put the dynamic loading function in a seperate file, rather than bundling with the testapi

Reply:
I plan to move things around as I am tackling usablity of this package later.